src/main/java/beans/menus/EntityListResponse.java

```diff
@@ -0,0 +1,133 @@
+package beans.menus;
+
+import io.swagger.annotations.ApiModel;
+import org.commcare.modern.session.SessionWrapper;
+import org.commcare.suite.model.Action;
+import org.commcare.suite.model.Detail;
+import org.commcare.suite.model.DetailField;
+import org.commcare.suite.model.EntityDatum;
+import org.commcare.util.cli.EntityDetailSubscreen;
+import org.commcare.util.cli.EntityScreen;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.xpath.XPathException;
+import util.SessionUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Vector;
+
+/**
+ * Created by willpride on 4/13/16.
+ */
+@ApiModel("Entity List Response")
+public class EntityListResponse extends MenuSessionBean {
+    private Entity[] entities;
+    private DisplayElement action;
+    private Style[] styles;
+
+    public EntityListResponse(){}
+
+    public EntityListResponse(EntityScreen nextScreen) {
+        SessionWrapper session = nextScreen.getSession();
+        Detail shortDetail = nextScreen.getShortDetail();
+        EvaluationContext ec = session.getEvaluationContext();
+        Vector<TreeReference> references = ec.expandReference(((EntityDatum)session.getNeededDatum()).getNodeset());
+        processTitle(session);
+        processEntities(nextScreen, references, ec);
+        processStyles(shortDetail);
+        processActions(shortDetail, nextScreen.getSession());
+    }
+
+    private void processTitle(SessionWrapper session) {
+        setTitle(SessionUtils.getBestTitle(session));
+    }
+
+    private void processEntities(EntityScreen screen, Vector<TreeReference> references, EvaluationContext ec) {
+        entities = new Entity[references.size()];
```

Comment:
where is this variable declared (might be missing something obvious)? also it doesn't seem to be used